benchmarks/callbacks/benchmark_SequenceOfRegionTimer.cpp

```diff
@@ -0,0 +1,149 @@
+        Kokkos::Tools::popRegion();
+
+        Kokkos::Tools::pushRegion("region B");
+        Kokkos::Tools::popRegion();
+
+        //! Add some regions that won't match, it's just noise.
+        for(unsigned short int irep = 0; irep < state.range(0); ++irep) {
+            Kokkos::Tools::pushRegion("region " + std::to_string(irep));
+            Kokkos::Tools::popRegion();
+        }
+
+        Kokkos::Tools::pushRegion("region C");
+        Kokkos::Tools::popRegion();
+
+        Kokkos::Tools::pushRegion("region D");
+        Kokkos::Tools::popRegion();
+    }
+};
+
+class SequenceOfRegionTimerBenchmark : public ::benchmark::Fixture, public MyFixture
+{
+public:
+    using matcher_t        = Kokkos::utils::callbacks::EventNameMatcher;
+    using focused_t        = Kokkos::utils::callbacks::SequenceOfRegionTimer<matcher_t>;
+    using region_timer_t   = typename focused_t::region_timer_t;
+    using region_matcher_t = typename region_timer_t::matcher_t;
+
+public:
+    void SetUp(::benchmark::State&) override
+    {
+        focused_t::initialize(std::vector{
+            region_matcher_t{{{"region A"}}},
+            region_matcher_t{{{"region B"}}},
+            region_matcher_t{{{"region C"}}},
+            region_matcher_t{{{"region D"}}}
+        });
+    }
+
+    void TearDown(const ::benchmark::State&) override {
+        focused_t::finalize();
+    }
+
+    //! Reset timers and run @ref MyFixture::run.
+    void run(const ::benchmark::State& state) const
+    {
+        focused_t::reset();
+        MyFixture::run(state);
+    }
+};
+
+class ManagerBenchmark : public ::benchmark::Fixture, public MyFixture
+{
+public:
+    using matcher_t      = Kokkos::utils::callbacks::EventNameMatcher;
+    using region_timer_t = Kokkos::utils::callbacks::RegionTimerListener<matcher_t>;
+
+public:
+    void SetUp(::benchmark::State&) override
+    {
+        Kokkos::utils::callbacks::Manager::initialize();
+
+        for(const auto& name : {"region A", "region B", "region C", "region D"})
+        {
+            auto timer = std::make_shared<region_timer_t>(name);
+            Kokkos::utils::callbacks::Manager::register_listener(timer);
+            timers.push_back(std::move(timer));
+        }
+    }
+
+    void TearDown(const ::benchmark::State&) override
+    {
+        for(const auto& timer : timers)
+            Kokkos::utils::callbacks::Manager::register_listener(timer);
+
+        Kokkos::utils::callbacks::Manager::finalize();
+    }
+
+    //! Reset timers and run @ref MyFixture::run.
+    void run(const ::benchmark::State& state) const
+    {
+        for(auto& timer : timers)
+            timer->reset();
+        MyFixture::run(state);
+    }
+
+private:
+    std::vector<std::shared_ptr<region_timer_t>> timers;
+};
+
+void custom_arguments(benchmark::internal::Benchmark* b) {
+    b->Arg(0)->Arg(25)->Arg(250);
+}
+
+BENCHMARK_DEFINE_F(SequenceOfRegionTimerBenchmark, 4)(::benchmark::State& state) {
```

Comment:
```suggestion
BENCHMARK_DEFINE_F(SequenceOfRegionTimerBenchmark, DurationsTest)(::benchmark::State& state) {
```
(not sure though)

Reply:
Adjusted to a nice name :heavy_check_mark: